swarmrl/engine/espresso.py

```diff
@@ -428,6 +427,95 @@ def add_confining_walls(self, wall_type: int):
         # the wall itself has no radius, only the particle radius counts
         self.colloid_radius_register.update({wall_type: 0.0})
 
+    def add_maze(
+        self,
+        maze_wall_start_point: pint.Quantity,
+        maze_wall_end_point: pint.Quantity,
+        maze_type: int,
+        wall_thickness: pint.Quantity,
+    ):
+        """
+        User defined walls will interact with particles through WCA.
+        Is NOT communicated to the interaction models, though.
+        The walls have a large height resulting in 2D-walls in a 2D-simulation.
+        The actual height adapts to the chosen box size.
+        The shape of the underlying constraint is a square.
+
+        Parameters
+        ----------
+        maze_wall_start_point : pint.Quantity
+        list of lists (len()=2) with wall coordinates
```

Comment:
it's probably easier for everyone if this is a (n,2) np.array instead of a list of lists